website/content/docs/running-staticcheck/cli/build-tags/index.md

```diff
@@ -0,0 +1,167 @@
+
+However, the number of build configurations you care about is probably much smaller.
+Your software probably support 2-3 operating systems on 1-2 architectures,
+and maybe has a debug and a release build.
+This makes for a lot fewer combinations that need to be checked.
+These are probably the same combinations you're already checking in CI, too, by running their tests.
+This will become useful in a bit.
+
+### The `-merge` flag
+
+Using the `-merge` flag, Staticcheck can merge the results of multiple runs.
+It decides on a per-check basis whether any run or all runs have to have reported an issue for it to be valid.
+It also takes into consideration which files were checked by which run, to reduce false negatives.
+
+In order to use `-merge`, the runs to be merged have to use the `-f binary` flag.
+This outputs results in a binary format containing all information required by `-merge`.
+When using `-merge`, arguments are interpreted as file names instead of import paths, so that `staticcheck -merge file1 file2` will read the files `file1` and `file2`, which must contain the output of `staticcheck -f binary` runs, and merge them.
+
+```terminal
+$ GOOS=linux staticcheck -f binary >file1
+$ GOOS=windows staticcheck -f binary >file2
+$ staticcheck -merge file1 file2
+...
+```
+
+
+Alternatively, if no arguments are passed, `staticcheck` will read from standard input instead.
+This allows for workflows like
+
+```
+(
+  GOOS=linux staticcheck -f binary
+  GOOS=windows staticcheck -f binary
+) | staticcheck -merge
+```
+
+This multi-step workflow of generating per-run output and merging it makes it possible to run Staticcheck on different systems before merging the results, which might be especially required when using cgo.
+
+### The `-matrix` flag
+
+With the `-matrix` flag, you can instruct Staticcheck to check multiple build configurations at once and merge the results.
+In other words, it automates running Staticcheck multiple times and merging results afterwards.
+This is useful when all configurations can be checked on a single system, for example because you don't use cgo.
+
+When using the `-matrix` flag, Staticcheck reads a build matrix from standard input.
+The build matrix uses a line-based format, where each line specifies a build name, environment variables and command-line flags.
+Each line is of the format `<name>: [flags and environment...]`, for example `linux-debug: GOOS=linux -tags=debug`.
+Environment variables and flags get passed to `go` when Staticcheck analyzes code, so you can use all flags that `go` supports, such as `-tags` or `-gcflags`, although few flags other than `-tags` are really useful.
+
+<details>
+<summary>Syntax rules of build matrices</summary>
+Build names may consist of Unicode numbers, Unicode letters, and underscores.
+Environment variable names may consist of a-z, A-Z, 0-9, and underscores.
+Flag names may consist of a-z, 0-9, underscores, and dashes, and they must begin with a dash.
```

Comment:
On a few of my projects, we use `--flag` instead of `-flag` for visibility and consistency with some GNU-ish tools, so I think it would be useful to mention explicitly, if the two-dash notation is supported or not.